framework/src/modules/chain/blocks/utils.js

```diff
@@ -112,66 +31,44 @@ const parseStorageObjToLegacyObj = block => {
  * @returns {Object} cb.rows - List of blocks
  */
 // eslint-disable-next-line class-methods-use-this
-const loadBlocksDataWS = async (storage, filter, tx) => {
-	const params = { limit: filter.limit || 1 };
-
-	// FIXME: filter.id is not used
-	if (filter.id && filter.lastId) {
-		throw new Error('Invalid filter: Received both id and lastId');
+const loadBlocksFromLastBlockId = async (storage, lastBlockId, limit) => {
+	if (!lastBlockId) {
+		throw new Error('lastBlockId needs to be specified');
 	}
-	if (filter.id) {
-		params.id = filter.id;
-	} else if (filter.lastId) {
-		params.lastId = filter.lastId;
+	if (!limit) {
+		throw new Error('Limit needs to be specified');
 	}
 
 	// Get height of block with supplied ID
-	const rows = await storage.entities.Block.get(
-		{ id: filter.lastId || null },
-		{ limit: params.limit },
-		tx,
+	const [lastBlock] = await storage.entities.Block.get(
+		{ id: lastBlockId },
+		{ limit },
 	);
-	if (!rows.length) {
+	if (!lastBlock) {
 		throw new Error('Invalid lastBlockId requested');
 	}
 
-	const height = rows.length ? rows[0].height : 0;
-	// Calculate max block height for database query
-	const realLimit = height + (parseInt(filter.limit, 10) || 1);
+	const lastBlockHeight = lastBlock.height;
 
-	params.limit = realLimit;
-	params.height = height;
-
-	const mergedParams = { ...filter, ...params };
-	const queryFilters = {};
-
-	if (!mergedParams.id && !mergedParams.lastId) {
-		queryFilters.height_lt = mergedParams.limit;
-	}
-
-	if (mergedParams.id) {
-		queryFilters.id = mergedParams.id;
-	}
+	// Calculate max block height for database query
+	const fetchUntilHeight = lastBlockHeight + limit;
 
-	if (mergedParams.lastId) {
-		queryFilters.height_gt = mergedParams.height;
-		queryFilters.height_lt = mergedParams.limit;
-	}
-	const blockRows = await storage.entities.Block.get(
-		queryFilters,
-		{
-			extended: true,
-			limit: null,
-			sort: ['height'],
-		},
-		tx,
-	);
+	const filter = {
+		height_gt: lastBlockHeight,
+		height_lt: fetchUntilHeight,
```

Comment:
I think you want less than or equal to here

```suggestion
		height_lte: fetchUntilHeight,
```